### DISCUSSION TITLE
livewire with yajra actions integration

### DISCUSSION THREAD

**Discussion by fdelvals** [livewire/livewire#2889] [1. Help]
Hello,
I have this code in controller to fetch data from model and into datatable:

` $acquisitions = InventoryAcquisitionView::all();

      return Datatables::of($acquisitions)
                    ->addIndexColumn()
                    ->addColumn('action', function($row){
                          return '<button **wire:click="confirmDelete()"**><i class="glyphicon glyphicon-edit"></i> Edit</button>';
                    })
                    ->rawColumns(['action'])
                    ->make(true);`

In view, i use jquery to populate yajra datatable

      `var table = $('.data-table').DataTable({

                    deferLoading: true,
                    drawCallback: function(settings) {
                        if (window.livewire) {
                            window.livewire.rescan();
                        }
                    },
            ajax: "{{ route('data') }}",
            columns: [
                ....
            ]
        });
    });`

The prolem is that livewire functions (marked with ** **) do not activate the methods. 
I read that the window.livewire.rescan() method would activate them, but no luck.

Does anyone have any idea / working code to se me up in the right direction?

thanks

**Comment by binaryweavers** [comment]
`Livewire.rescan()` would not activate the `livewire directives` such as `wire:click, wire:model` etc. 
`rescan` looks for new livewire components based on `wire:initial-data` when found it initializes the components. directive initialization is part of component initialization.
So the code you were trying won't work they way you are expecting.

**Reply by fdelvals** [reply]
is there any alternative?

I ended up switching to jquery because nothing worked

**Comment by ousid** [comment]
Hey, @fdelvals 

Check out [Livewire Datatable](https://github.com/MedicOneSystems/livewire-datatables) by @marksalmon It's a really good one.

**Reply by binaryweavers** [reply]
I was also looking for this to suggest. Thanks 👍🏼
